File: src/main/java/com/clinichub/entity/Prescription.java
package com.clinichub.entity;

import jakarta.persistence.*;
import lombok.*;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;

@Entity
@Table(name = "prescription")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Prescription {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "visit_id", nullable = false)
    private Visit visit;

    @Column(nullable = false)
    private Integer copies;

    @Column(name = "usage_text", length = 255)
    private String usageText;

    @Column(length = 20, nullable = false)
    private String status;

    @Column(columnDefinition = "longtext")
    private String remark;

    @Column(name = "created_at", nullable = false)
    private java.time.LocalDateTime createdAt;

    @OneToMany(mappedBy = "prescription", fetch = FetchType.LAZY)
    private List<PrescriptionItem> items;
}

File: flutter_app/lib/models/prescription.dart
import 'prescription_item.dart';
import 'visit.dart';

class Prescription {
  final int id;
  final Visit visit;
  final int copies;
  final String? usageText;
  final String status;
  final String? remark;
  final String createdAt;
  final List<PrescriptionItem>? items;

  Prescription({
    required this.id,
    required this.visit,
    required this.copies,
    this.usageText,
    required this.status,
    this.remark,
    required this.createdAt,
    this.items,
  });

  factory Prescription.fromJson(Map<String, dynamic> json) => Prescription(
    id: json['id'],
    visit: Visit.fromJson(json['visit']),
    copies: json['copies'],
    usageText: json['usageText'],
    status: json['status'],
    remark: json['remark'],
    createdAt: json['createdAt'],
    items: json['items'] != null
        ? (json['items'] as List).map((e) => PrescriptionItem.fromJson(e)).toList()
        : null,
  );
}
